managed/resource_cluster.go

```diff
@@ -474,6 +510,30 @@ func validateCredentials(credentials Credentials) bool {
 
 }
 
+func createCmkSpec(plan Cluster) (cmkSpec *openapiclient.CMKSpec, cmkSpecOK bool, errorMessage string) {
+	cmkSpecOK = true // Assume CMKSpec is valid unless an error occurs
+	cmkProvider := plan.CMKSpec.ProviderType.Value
+	cmkSpec = openapiclient.NewCMKSpec(cmkProvider)
+
+	if strings.ToLower(cmkProvider) == "gcp" {
```

Comment:
Maybe check `!= aws`, any other thing string should not be allowed. You can also consider using a validator like this: https://github.com/yugabyte/terraform-provider-ybm/blob/main/managed/resource_cluster.go#LL67C5-L67C89